### DISCUSSION TITLE
GPUI wish list

### DISCUSSION THREAD

**Discussion by krstp** [zed-industries/zed#20140] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

A few wishes:

- I am trying to get a grip around GPUI. Venturing into `examples/` folder (https://github.com/zed-industries/zed/tree/main/crates/gpui/examples) I tried to check some of the samples. As such `animation.py` does not seem to cause any animation, just a black `hello` on green square. What I am reporting here is I suppose a bug of examples code.

- Additionally trying different examples, this is the message I get:
```
2024-10-19 19:17:04.236 gpui[35381:2487415] +[IMKClient subclass]: chose IMKClient_Legacy
2024-10-19 19:17:04.236 gpui[35381:2487415] +[IMKInputSession subclass]: chose IMKInputSession_Legacy
2024-10-19 19:17:04.245 gpui[35381:2487415] NSKeyBindingManager: Bad key binding selector for '^@' = 'noop'
```

- Would be great if `gpui` (https://www.gpui.rs/#examples) deserved its own repository. It has its own website, but the only code seems to live as Zed's crate 🤔 

- gif viewer also seems to be missing some creates out of the box and not stated dependency, i.e.
```
28 |     env_logger::init();
   |     ^^^^^^^^^^ use of undeclared crate or module `env_logger`
```
- Overall my wish if if GPUI could be more self contained with workable sample code. Would be great to pick up on it

### Environment

MacOS

### If applicable, add mockups / screenshots to help explain present your vision of the feature

See text above.

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by bowenxuuu** [comment]
Agreed!
For a long time, the Rust community has lacked a powerful native GUI framework. I hope that `GPUI` can become the most powerful and general-purpose native GUI framework in Rust, just like `Qt` in C++.

**Comment by mrnugget** [comment]
> * Additionally trying different examples, this is the message I get:

Seems to be a macOS problem: https://github.com/rust-windowing/winit/issues/3931

**Comment by mrnugget** [comment]
> * As such `animation.py` does not seem to cause any animation, just a black `hello` on green square. What I am reporting here is I suppose a bug of examples code.

Works here:

```
$ cd crates/gpui
$ cargo run --example animation
```
![screenshot-2024-10-29-11 25 52](https://github.com/user-attachments/assets/21621101-bd63-46ac-a29e-e06f65ba4f70)

**Comment by krstp** [comment]
Did you run it out of the box on fresh env or being the zed developer, which I think you are, did you run it in your pre-existing development env? What I meant, it does not work on fresh install, imho some extra crates need to be added. When I find a minute I will try to look more into it.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support

**Comment by cosoc** [comment]
世界不缺一个编辑器,但是世界一定缺一个好的GUI库,建议完全独立成正式项目和单独维护.

**Reply by krstp** [reply]
Agreed. And well... 2yrs in the making and still not much progress on the separation front. IMHO: stuff like this keeps Rust away from full adoption.

PS. To be honest I have not been up to date with latest developments, so I might be lagging on current reality check. Needless to say, seeing people still post on the subject tells me there has not been much progress ;)

**Reply by Jimmy-Z** [reply]
there's gpui-ce, check it out, maybe that's what you want
